### PR TITLE
Feature: Merge imports from the same package

### LINKED ISSUE #348: Feature: Merge imports from the same package during import organization
#### Feature: Merge imports from the same package during import organization

There are cases that there might be multiple imports in a file from the same package/file. For example:

```ts
import { Khaleesi } from 'got';
import { JohnSnow } from 'got';
```

What could happen after import organization is that the result would look as follows:

```ts
import { JohnSnow, Khaleesi } from 'got';
```

#### Reason

This would group multiple lines from the same package into a single import thus improving readability and organization or the improts. It is much better to see all the improts from the same module in a single import line.

### PR DESCRIPTION
**Pull request by shobhitg**
Merge imports from the same package during import organization

- Closes #348

#### Description

Merge imports from the same package during import organization

### CODE REVIEW

**Comment by shobhitg** [comment]
I have no clue why `travis-ci` is unhappy.

**Comment by shobhitg** [comment]
It seems like `src/import-organizer/import-manager.ts` has 2 space indents.

But the project's workspace file had tabSize set to 4.

But yet in another file I am seeing 4 space indents.

Is tabSize supposed to be 2 or 4 ?

**Comment by buehler** [comment]
Actually, I switched to tabSize 2. (the `.editorconfig` should help you with that)

File: src/import-organizer/import-manager.ts [hunk 1]
```diff
@@ -153,9 +153,20 @@ export default class ImportManager {
...
           const defaultSpec = actImport.defaultAlias;
+          const libraryAlreadyImported = keep.find(d => d.libraryName === actImport.libraryName);
           if (actImport.specifiers.length ||
             (!!defaultSpec && this._parsedDocument.nonLocalUsages.indexOf(defaultSpec) >= 0)) {
-            keep.push(actImport);
+            if (libraryAlreadyImported) {
+              if (actImport.defaultAlias) {
+                (<NamedImport>libraryAlreadyImported).defaultAlias = actImport.defaultAlias;
+              }
+              (<NamedImport>libraryAlreadyImported).specifiers = [
```
Comment (buehler) [inline src/import-organizer/import-manager.ts]: Clever :-)

**Pull request merged**